src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/config/PluginSettings.java

```diff
@@ -271,4 +294,26 @@ private void loadMetricsDBFilesCleanupEnabled() {
       shouldCleanupMetricsDBFiles = true;
     }
   }
+
+  private void loadBatchMetricsRetentionPeriodFromConfig() {
+    if (!settings.containsKey(BATCH_METRICS_RETENTION_PERIOD)) {
+      return;
+    }
+
+    long parsedRetentionPeriod;
+    try {
+      parsedRetentionPeriod = Long.parseLong(settings.getProperty(BATCH_METRICS_RETENTION_PERIOD));
+      if (parsedRetentionPeriod <= 0 || parsedRetentionPeriod > BATCH_METRICS_RETENTION_PERIOD_LIMIT) {
+        throw new InvalidParameterException();
```

Comment:
We are throwing an exception here but maybe we should just log an error and use default values ?

Reply:
Joydeep, if any exception is thrown here, the batchMetricsRetentionPeriodMinutes value will not have been modified (since it is in the last line of the try block), and so the default value will be retained. Additionally, the method itself catches the thrown Exception and logs it.

Reply:
Sid, Sruti's pr is only about disabling the reader if the config file cannot be found. It was not about disabling the reader if values cannot be correctly parsed from the config file -- using the default value for these cases is the desired behavior.